src/main/kotlin/uk/gov/gdx/datashare/models/DeathNotificationDetails.kt

```diff
@@ -52,4 +57,72 @@ data class DeathNotificationDetails(
 
   @Schema(description = "Whether the deceased was retired", required = false, example = "false")
   val retired: Boolean? = null,
-)
+) {
+  companion object {
+    fun fromLevDeathRecord(enrichmentFields: List<String>, levDeathRecord: LevDeathRecord): DeathNotificationDetails {
+      with(levDeathRecord) {
+        return DeathNotificationDetails(
+          enrichmentFields = enrichmentFields,
+          registrationDate = if (enrichmentFields.contains("registrationDate")) date else null,
+          firstNames = if (enrichmentFields.contains("firstNames")) deceased.forenames else null,
+          lastName = if (enrichmentFields.contains("lastName")) deceased.surname else null,
+          sex = if (enrichmentFields.contains("sex")) deceased.sex else null,
+          dateOfDeath = if (enrichmentFields.contains("dateOfDeath")) deceased.dateOfDeath else null,
+          dateOfBirth = if (enrichmentFields.contains("dateOfBirth")) deceased.dateOfBirth else null,
+          birthPlace = if (enrichmentFields.contains("birthPlace")) deceased.birthplace else null,
+          deathPlace = if (enrichmentFields.contains("deathPlace")) deceased.deathplace else null,
+          maidenName = if (enrichmentFields.contains("maidenName")) deceased.maidenSurname else null,
+          occupation = if (enrichmentFields.contains("occupation")) deceased.occupation else null,
+          retired = if (enrichmentFields.contains("retired")) deceased.retired else null,
+          address = if (enrichmentFields.contains("address")) deceased.address else null,
+        )
+      }
+    }
+  }
+}
+
+class DeathNotificationDetailsSerializer :
+  StdSerializer<DeathNotificationDetails>(DeathNotificationDetails::class.java) {
+  override fun serialize(value: DeathNotificationDetails?, gen: JsonGenerator?, provider: SerializerProvider?) {
+    gen?.writeStartObject()
+    value?.run {
+      if (enrichmentFields.contains("registrationDate")) {
+        gen?.writeNullableStringField("registrationDate", registrationDate?.toString())
+      }
+      if (enrichmentFields.contains("firstNames")) {
+        gen?.writeNullableStringField("firstNames", firstNames)
+      }
+      if (enrichmentFields.contains("lastName")) {
+        gen?.writeNullableStringField("lastName", lastName)
+      }
+      if (enrichmentFields.contains("sex")) {
+        gen?.writeNullableStringField("sex", sex?.jsonName)
+      }
+      if (enrichmentFields.contains("dateOfDeath")) {
+        gen?.writeNullableStringField("dateOfDeath", dateOfDeath?.toString())
```

Comment:
Do these serialize as we'd expect or should we format the date ourselves?

Reply:
For example would it be better to do this date->string/json mapping within our writeNullableStringField bit instead?

Reply:
Yeah this comes out fine. It gets messy when you start adding times, but date works